From 6807f610931a5909c19565a5f842548342ac302d Mon Sep 17 00:00:00 2001
From: "ji, zhenlong z" <zhenlong.z.ji@intel.com>
Date: Fri, 10 Sep 2021 09:13:41 +0800
Subject: [PATCH] Disable the Status Bar and grant runtime permissions by
 default

Based on the customer requirement, we need to remove the
status bar and disable the permission grant pop-up
window.

Signed-off-by: ji, zhenlong z <zhenlong.z.ji@intel.com>
---
 .../android/server/pm/permission/PermissionManagerService.java  | 2 ++
 services/core/java/com/android/server/wm/DisplayPolicy.java     | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/pm/permission/PermissionManagerService.java b/services/core/java/com/android/server/pm/permission/PermissionManagerService.java
index 8d2363b6e831..0b50e353956f 100644
--- a/services/core/java/com/android/server/pm/permission/PermissionManagerService.java
+++ b/services/core/java/com/android/server/pm/permission/PermissionManagerService.java
@@ -2845,6 +2845,8 @@ public class PermissionManagerService extends IPermissionManager.Stub {
                             + friendlyName);
                 }
 
+                grant = GRANT_INSTALL;
+
                 if (grant != GRANT_DENIED) {
                     if (!ps.isSystem() && ps.areInstallPermissionsFixed() && !bp.isRuntime()) {
                         // If this is an existing, non-system package, then
diff --git a/services/core/java/com/android/server/wm/DisplayPolicy.java b/services/core/java/com/android/server/wm/DisplayPolicy.java
index 29881cc761b2..4d8857d2fb0b 100644
--- a/services/core/java/com/android/server/wm/DisplayPolicy.java
+++ b/services/core/java/com/android/server/wm/DisplayPolicy.java
@@ -3621,7 +3621,7 @@ public class DisplayPolicy {
             StatusBarManagerInternal statusBar = getStatusBarManagerInternal();
             if (statusBar != null) {
                 final int displayId = getDisplayId();
-                statusBar.setDisableFlags(displayId, visibility & StatusBarManager.DISABLE_MASK,
+                statusBar.setDisableFlags(displayId, StatusBarManager.DISABLE_MASK,
                         cause);
                 if (transientState.first.length > 0) {
                     statusBar.showTransient(displayId, transientState.first);
-- 
2.25.1

